src/core/thread/mesh_forwarder.cpp

```diff
@@ -815,66 +815,65 @@ Mac::TxFrame *MeshForwarder::HandleFrameRequest(Mac::TxFrames &aTxFrames)
-    // Determine Header IE entries
+    // Determine frame version and Header IE entries
 
-#if OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
+    neighbor = Get<NeighborTable>().FindNeighbor(aTxFrameInfo.mAddrs.mDestination);
 
-#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
-    if ((aMessage != nullptr) && aMessage->IsTimeSync())
+    if (neighbor == nullptr)
     {
-        aTxFrameInfo.mAppendTimeIe = true;
-        aTxFrameInfo.mVersion      = Mac::Frame::kVersion2015;
     }
-#endif
 #if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
-    if (Get<Mac::Mac>().IsCslEnabled() &&
-        !(aMessage != nullptr && aMessage->IsMleCommand(Mle::kCommandDiscoveryRequest)))
+    else if (Get<Mac::Mac>().IsCslEnabled())
     {
         aTxFrameInfo.mAppendCslIe = true;
         aTxFrameInfo.mVersion     = Mac::Frame::kVersion2015;
     }
 #endif
-
-    aTxFrameInfo.mEmptyPayload = (aMessage == nullptr) || (aMessage->GetLength() == 0);
-
-#endif // OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT
-
-#if (OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE) || \
-    OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
-
-    //- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
-    // Determine frame version
-
-    if (aTxFrameInfo.mVersion == Mac::Frame::kVersion2006)
-    {
-        const Neighbor *neighbor = Get<NeighborTable>().FindNeighbor(aTxFrameInfo.mAddrs.mDestination);
-
 #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
-        if ((neighbor != nullptr) && Get<ChildTable>().Contains(*neighbor) &&
-            static_cast<const Child *>(neighbor)->IsCslSynchronized())
-        {
-            aTxFrameInfo.mVersion = Mac::Frame::kVersion2015;
-        }
+    else if ((Get<ChildTable>().Contains(*neighbor) && static_cast<const Child *>(neighbor)->IsCslSynchronized()))
+    {
+        aTxFrameInfo.mVersion = Mac::Frame::kVersion2015;
+    }
 #endif
 #if OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
-        if ((neighbor != nullptr) && neighbor->IsEnhAckProbingActive())
-        {
-            aTxFrameInfo.mVersion = Mac::Frame::kVersion2015;
-        }
+    else if (neighbor->IsEnhAckProbingActive())
+    {
+        aTxFrameInfo.mVersion = Mac::Frame::kVersion2015;
+    }
 #endif
+
+#endif // (OPENTHREAD_FTD && OOPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE) || OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+       // || OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
+
+#if OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
+    if ((aMessage != nullptr) && aMessage->IsTimeSync())
+    {
+        aTxFrameInfo.mAppendTimeIe = true;
+        aTxFrameInfo.mVersion      = Mac::Frame::kVersion2015;
     }
+#endif
 
-#endif // OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE || OPENTHREAD_CONFIG_MLE_LINK_METRICS_INITIATOR_ENABLE
+    aTxFrameInfo.mEmptyPayload = (aMessage == nullptr) || (aMessage->GetLength() == 0);
```

Comment:
How about moving `aTxFrameInfo.mEmptyPayload = (aMessage == nullptr) || (aMessage->GetLength() == 0);` outside of `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT` block?

I know it was before within the same block but not sure if it need to be. What do you think?

Reply:
The field `mEmptyPayload` is only present when `OPENTHREAD_CONFIG_MAC_HEADER_IE_SUPPORT` is enabled. The field is used to determine whether a termination IE is needed.

Reply:
I see. Got it. That's why it is within that block.